fdbserver/workloads/Cycle.actor.cpp

```diff
@@ -51,17 +65,57 @@ struct CycleWorkload : TestWorkload {
 		keyPrefix = unprintable(getOption(options, "keyPrefix"_sr, LiteralStringRef("")).toString());
 		traceParentProbability = getOption(options, "traceParentProbability"_sr, 0.01);
 		minExpectedTransactionsPerSecond = transactionsPerSecond * getOption(options, "expectedRate"_sr, 0.7);
+		if constexpr (MultiTenancy) {
+			ASSERT(g_network->isSimulated());
+			auto k = g_simulator.authKeys.begin();
+			this->tenant = getOption(options, "tenant"_sr, "CycleTenant"_sr);
+			// make it comfortably longer than the timeout of the workload
+			auto currentTime = uint64_t(lround(g_network->timer()));
+			this->token.algorithm = authz::Algorithm::ES256;
+			this->token.issuedAtUnixTime = currentTime;
+			this->token.expiresAtUnixTime = currentTime + uint64_t(std::lround(getCheckTimeout())) + uint64_t(100);
```

Comment:
100 seems to be just an arbitrary number to keep token valid well after the test times out.
Wouldn't timeout take precedence and fail the workload anyway if `testDuration` becomes longer than the timeout?

Reply:
I think `testDuration` is the timeout length if I understand correctly, and 100 was chosen here so that it would expire after the test timed out. However, if somebody passes in a different value for `testDuration`, then the relationship between expiration and test timeout could flip. To avoid that, we could change the constant value `100` to be some number based on (but larger than) `testDuration`.

Reply:
Understood. Done!